frontends/web/src/routes/settings/manage-accounts.tsx

```diff
@@ -264,46 +287,51 @@ class ManageAccounts extends Component<Props, State> {
             <View fullscreen={false}>
               <ViewContent>
                 <WithSettingsTabs deviceIDs={deviceIDs} hideMobileMenu hasAccounts={hasAccounts}>
-                  <>
-                    <Button
-                      className={style.addAccountBtn}
-                      primary
-                      onClick={() => route('/add-account', true)}>
-                      {t('addAccount.title')}
-                    </Button>
-
-                    {
-                      accountsByKeystore.map(keystore => (<React.Fragment key={keystore.keystore.rootFingerprint}>
+                  <Button
+                    className={style.addAccountBtn}
+                    primary
+                    onClick={() => route('/add-account', true)}>
+                    {t('addAccount.title')}
+                  </Button>
+                  {
+                    accountsByKeystore.map(keystore => (
+                      <React.Fragment key={keystore.keystore.rootFingerprint}>
                         <p>{keystore.keystore.name}</p>
                         <div className="box slim divide m-bottom-large">
-                          { this.renderAccounts(keystore.accounts) }
+                          {this.renderAccounts(keystore.accounts)}
                         </div>
-                      </React.Fragment>))
-                    }
-                    { accounts.length === 0 ? t('manageAccounts.noAccounts') : null }
-
+                      </React.Fragment>
+                    ))
+                  }
+                  {currentlyEditedAccount && (
                     <Dialog
-                      open={!!(editAccountCode)}
-                      onClose={() => this.setState({ editAccountCode: undefined, editAccountNewName: '', editErrorMessage: undefined })}
+                      open={!!(currentlyEditedAccount)}
+                      onClose={() => this.setState({ currentlyEditedAccount: undefined })}
                       title={t('manageAccounts.editAccountNameTitle')}>
                       <form onSubmit={this.updateAccountName}>
                         <Message type="error" hidden={!editErrorMessage}>
                           {editErrorMessage}
                         </Message>
                         <Input
-                          onInput={e => this.setState({ editAccountNewName: e.target.value })}
-                          value={editAccountNewName} />
+                          onInput={e => this.setState({ currentlyEditedAccount: { ...currentlyEditedAccount, name: e.target.value } })}
+                          value={currentlyEditedAccount.name}
+                        />
+                        {watchonly &&
```

Comment:
is the fragment needed here or would that work?

```suggestion
                        {watchonly && this.renderWatchOnlyToggle()}
```